sql/src/main/java/io/crate/metadata/settings/SessionTransportableInfo.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.metadata.settings;
+
+import io.crate.metadata.SearchPath;
+import io.crate.metadata.settings.session.SessionSettingRegistry;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public final class SessionTransportableInfo implements Writeable {
+
+    private String userName;
+    private Map<String, String> sessionSettings;
+    private SearchPath searchPath;
+
+
+    public static SessionTransportableInfo fromStream(StreamInput in) throws IOException {
+        final String userName = in.readString();
+
+        final int numOfSettings = in.readVInt();
+        Map<String, String> sessionSettings = new HashMap<>();
+        for (int i = 0; i < numOfSettings; i++) {
+            sessionSettings.put(in.readString(), in.readString());
+        }
+        return new SessionTransportableInfo(userName, sessionSettings);
+    }
+
+    public SessionTransportableInfo(String userName,
+                                    Map<String, String> sessionSettings) {
+        this.userName = userName;
+        this.sessionSettings = sessionSettings;
+        this.searchPath = SearchPath.createSearchPathFrom(
+            sessionSettings.get(SessionSettingRegistry.SEARCH_PATH_KEY));
```

Comment:
Imho we should either treat all settings explicitly or none, but not mixing them (like searchPath is a direct getter while the other are not). I tend to use them all explicitly.
Also we should keep the types of each settings and not convert them back and forth to/from String.
This could be done by implementing e.g. `getValueRaw` on the `SessionSetting`.